azure/services/agentpools/agentpools.go

```diff
@@ -19,7 +19,7 @@ package agentpools
 import (
 	"context"
 
-	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-02-01/containerservice"
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-03-01/containerservice"
```

Comment:
any chance I can persuade you to upgrade to `2021-05-01` 😃 the diff should be basically nothing. if there's any non-trivial diff, you can leave it at `2021-03-01`.

Reply:
ack 👍

Reply:
The one minor difference while moving it from 2021-03-01 - 2021-05-01 is 

[ListClusterAdminCredentials](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/15da9c80de36e90f3f564e9c75e928c8003b7614/azure/services/managedclusters/client.go#L69) api has an extra param serverFqdn in 2021-05-01. I have kept it to empty string.